server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamTests.java

```diff
@@ -208,23 +208,8 @@ public void testRolloverWithConflictingBackingIndexName() {
     public void testRolloverUpgradeToTsdbDataStream() {
         IndexMode indexMode = randomBoolean() ? IndexMode.STANDARD : null;
         DataStream ds = DataStreamTestHelper.randomInstance().promoteDataStream();
-        // Unsure index_mode=null
-        ds = new DataStream(
-            ds.getName(),
-            ds.getIndices(),
-            ds.getGeneration(),
-            ds.getMetadata(),
-            ds.isHidden(),
-            ds.isReplicated(),
-            ds.isSystem(),
-            ds.isAllowCustomRouting(),
-            indexMode,
-            ds.getLifecycle(),
-            ds.isFailureStore(),
-            ds.getFailureIndices(),
-            ds.rolloverOnWrite(),
-            ds.getAutoShardingEvent()
-        );
+        // Ensure index_mode=null
```

Comment:
So, here you change the comment from `unsure` to `ensure`. Honestly, I think this comment should be removed because I do not see what is its value. However, I think `ensure` is wrong because `indexMode` is set by:
```
IndexMode indexMode = randomBoolean() ? IndexMode.STANDARD : null
```

Also I think this could be rewritten simpler to:

```
new DataStream.Builder(DataStreamTestHelper.randomInstance()).setReplicated(false).setIndexMode(indexMode).build()
```